lib/utils/compose_ts.ts

```diff
@@ -179,13 +179,18 @@ async function mergeDevComposeOverlay(
 		logger.logInfo(
 			`Docker compose dev overlay detected (${devOverlayFilename}) - merging.`,
 		);
+		interface ComposeObj {
+			services?: object;
+		}
+		const yaml = await import('js-yaml');
+		const loadObj = (inputStr: string): ComposeObj =>
+			(yaml.load(inputStr) || {}) as ComposeObj;
 		try {
-			const yaml = await import('js-yaml');
-			const compose = yaml.load(composeStr);
-			const devOverlay = yaml.load(await fs.readFile(devOverlayPath, 'utf8'));
+			const compose = loadObj(composeStr);
+			const devOverlay = loadObj(await fs.readFile(devOverlayPath, 'utf8'));
 			// We only want to merge the services section
 			compose.services = { ...compose.services, ...devOverlay.services };
-			composeStr = yaml.dump(compose);
+			composeStr = yaml.dump(compose, { styles: { '!!null': 'empty' } });
```

Comment:
`{ styles: { '!!null': 'empty' } }` is what fixes issue #2226. This option was introduced in `js-yaml` v4.0.0, which is why this PR updates `js-yaml`.